arch/risc-v/src/esp32c3/esp32c3_start.c

```diff
@@ -136,6 +177,30 @@ uintptr_t g_idle_topstack = ESP32C3_IDLESTACK_TOP;
  * Private Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: setup_syscall_table
+ *
+ * Description:
+ *   Configure the syscall table used by the ROM code for calling C library
+ *   functions.
+ *   ESP32-C3 ROM code contains implementations of some of C library
+ *   functions. Whenever a function in ROM needs to use a syscall, it calls
+ *   a pointer to the corresponding syscall implementation defined in the
+ *   syscall_stub_table struct.
+ *
+ * Input Parameters:
+ *   None.
+ *
+ * Returned Value:
+ *   None.
+ *
+ ****************************************************************************/
+
+static void setup_syscall_table(void)
+{
+  syscall_table_ptr = &g_stub_table;
```

Comment:
is syscall_table_ptr located in rom region in ld script? If so, why can we change it value here? Could you explain more detail info and usage about syscall table.

Reply:
Yes, the basic idea of it is to force the ROM functions to use the system (NuttX) libc instead of using the internal newlibc implemented in the ROM code.

Reply:
if so why not locate g_stub_table in ld script directly?